go/app/infra.go

```diff
@@ -30,99 +29,65 @@ type ItemRepository interface {
 
 // itemRepository is an implementation of ItemRepository
 type itemRepository struct {
-	// fileName is the path to the JSON file storing items.
-	fileName string
+	// db is a database connection
+	db *sql.DB
 }
 
-// NewItemRepository creates a new itemRepository.
-func NewItemRepository() ItemRepository {
-	return &itemRepository{fileName: "./db/items.json"}
+// NewItemRepository connects db and creates a new itemRepository.
+func NewItemRepository(dbPath string) (ItemRepository, error) {
+	db, err := sql.Open("sqlite3", dbPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to the database: %w", err)
+	}
+	// if the table does not exist, create it
+	_, err = db.Exec(`
+		CREATE TABLE IF NOT EXISTS items (
+			id INTEGER PRIMARY KEY AUTOINCREMENT,
+			name TEXT NOT NULL,
+			category TEXT NOT NULL,
+			image_name TEXT NOT NULL
+		);
+	`)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create table: %w", err)
+	}
+	return &itemRepository{db: db}, nil
 }
 
 // Insert inserts an item into the repository.
 func (i *itemRepository) Insert(ctx context.Context, item *Item) error {
-	// STEP 4-2: add an implementation to store an item
-	// 既存データの読み込み
-	items, err := i.loadItems()
+	_, err := i.db.Exec(
+		"INSERT INTO items (name, category, image_name) VALUES (?, ?, ?)",
```

Comment:
We also have to create a separate `categories` table and use SQL joins to connect category and category ID